angel_system/activity_hmm/core.py

```diff
@@ -178,23 +201,23 @@ def analyze_current_state(self):
             Score indicating how much more likely it is that a step was skipped
             than no steps being skipped.
         """
-        log_prob1, Z_no_skip, X_, Z_no_skip_ = model.decode(X)
-        log_prob1 = model_skip.calc_log_prob_(X_, Z_no_skip_)
+        log_prob1, Z_no_skip, X_, Z_no_skip_ = self.noskip_model.decode(X)
```

Comment:
Use of `X` here is undefined. Did you mean `self.X`?

Reply:
Fixed (sorry, playing too fast and loose).